src/main/java/com/cloudbees/plugins/credentials/xml/XMLUtils.java

```diff
@@ -0,0 +1,101 @@
+package com.cloudbees.plugins.credentials.xml;
+
+import java.util.logging.Level;
+import java.util.logging.LogManager;
+import java.util.logging.Logger;
+import javax.annotation.Nonnull;
+import javax.xml.XMLConstants;
+import javax.xml.transform.Result;
+import javax.xml.transform.Source;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.sax.SAXSource;
+import javax.xml.transform.sax.SAXTransformerFactory;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+import org.xml.sax.XMLReader;
+import org.xml.sax.helpers.XMLReaderFactory;
+
+/**
+ * Utilities useful when working with various XML types.
+ */
+@Restricted(NoExternalUse.class)
```

Comment:
Why?  this is a brutal yank from core and is missing the copyright header.
we can just use the one from core (we already had - just ignore the restriction temporarily as it is removed in the latest master)
At the very least this should include a TODO to remove when the baseline is > whatever Fridays release will be.